core/controllers/classifier_test.py

```diff
@@ -152,8 +153,14 @@ def mock_get_classifier_training_job_by_id(_):
         config_property.set_value(
             'committer_id', ['moderator@example.com'])
 
-        with can_send_emails_ctx, can_send_feedback_email_ctx:
-            with fail_training_job:
+        with can_send_emails_ctx, can_send_feedback_email_ctx, (
+            self.swap(
```

Comment:
I'm not sure if you saw my other comment, but the only thing you should be swapping in these tests is the single function in mailgun_email_services.

Ideally EmailTestBase would handle that for you, so that inheriting EmailTestBase is sufficient and you don't need to manually do the swap each time. I can think of two possible approaches to investigate -- one is to dig into what happens when a `def test_...()` function is executed and see if you can find a way to overwrite the test harness framework to run those functions within a self.swap() context manager. The other is to monkeypatch mailgun_email_services.send_email_to_recipients by setting it to some other value in setUp, and setting it back to its original value in tearDown.